File: workflows/pythonSQL2in2out/code/src/main/python/graph/Source1.py
from pyspark.sql import SparkSession
from pyspark.sql.types import *
from pyspark.sql.functions import *
from ..config import Config
from ..prophecylibs import *

@Visual(id = "Source1", label = "Source1", x = 156, y = 198, phase = 0)
@UsesDataset(id = "16", version = 0)
def Source1(spark: SparkSession) -> Source:
    fabric = Config.fabricName
    out = None

    if fabric == "azdb":
        schemaArg = StructType([
            StructField("order_id", IntegerType(), False),
            StructField("customer_id", IntegerType(), False),
            StructField("order_status", StringType(), False),
            StructField("order_category", StringType(), False),
            StructField("order_date", StringType(), False),
            StructField("amount", DoubleType(), False)
        ])
        out = spark.read\
                  .format("csv")\
                  .schema(schemaArg)\
                  .option("sep", ",")\
                  .option("inferSchema", True)\
                  .load("dbfs:/Users/prophecy/eng/OrdersDatasetInput.csv")
    else:
        raise ValueError("The fabric %s is not supported" % fabric)

    return out

File: workflows/scalaSQL2in2out/code/src/main/scala/graph/Source1.scala
package graph

import org.apache.spark.sql.types._
import io.prophecy.libs._
import io.prophecy.libs.UDFUtils._
import io.prophecy.libs.Component._
import io.prophecy.libs.DataHelpers._
import io.prophecy.libs.SparkFunctions._
import io.prophecy.libs.FixedFileFormatImplicits._
import org.apache.spark.sql.ProphecyDataFrame._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import config.ConfigStore._
import graph._

@Visual(id = "Source1", label = "Source1", x = 164, y = 228, phase = 0)
object Source1 {

  @UsesDataset(id = "16", version = 0)
  def apply(spark: SparkSession): Source = {
    import spark.implicits._

    val fabric = Config.fabricName

    val out = fabric match {
      case "azdb" =>
        val schemaArg = StructType(
          Array(
            StructField("order_id",       IntegerType, false),
            StructField("customer_id",    IntegerType, false),
            StructField("order_status",   StringType,  false),
            StructField("order_category", StringType,  false),
            StructField("order_date",     StringType,  false),
            StructField("amount",         DoubleType,  false)
          )
        )
        spark.read
          .format("csv")
          .option("sep", ",")
          .schema(schemaArg)
          .load("dbfs:/Users/prophecy/eng/OrdersDatasetInput.csv")
          .cache()
      case _ => throw new Exception(s"The fabric '$fabric' is not handled")
    }

    out

  }

}
